clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    }),
+    UNKNOWN_ERROR_CODE(57, "The client received an unknown error code when processing the response. This may happen if " +
+            "the client library version is lower than the server library version",
+        new ApiExceptionBuilder() {
```

Comment:
@lindong28 : Ok, what you said about the approach in the patch makes sense. I was just wondering if we can achieve the same w/o introducing UnknownErrorCode. The alternative that I suggested also just needs 1 if/else check. Basically, you bump up the client request protocol, if the client is on the old protocol, the broker sends an existing retriable error code. Otherwise, the broker sends the new specific error code. So, the logic is about the same as your patch, but we don't need to introduce UNKNOWN_ERROR_CODE. This seems a bit simpler since we don't have to explain to the client developers the subtle difference btw UNKNOWN_ERROR_CODE and UNKNOWN.